src/main/java/edu/ksu/canvas/attendance/controller/CourseConfigurationController.java

```diff
@@ -93,6 +119,71 @@ public ModelAndView synchronizeWithCanvas(@PathVariable String sectionId) throws
         return page;
     }
 
+    @RequestMapping(value = "/{sectionId}/save", params = "pushGradesToCanvas", method = RequestMethod.POST)
+    public ModelAndView pushGradesToCanvas(@PathVariable Long sectionId, @ModelAttribute("courseConfigurationForm") @Valid CourseConfigurationForm classSetupForm, BindingResult bindingResult) throws NoLtiSessionException{
+        validator.validate(classSetupForm, bindingResult);
+        if (bindingResult.hasErrors()) {
+            ModelAndView page = new ModelAndView("/courseConfiguration");
+            List<String> errors = new ArrayList<>();
+            bindingResult.getFieldErrors().forEach(error -> errors.add(error.getCode()));
+            page.addObject("error", errors);
+            page.addObject("selectedSectionId", sectionId);
+            return page;
+        } else {
+            LOG.info("eid: " + canvasService.getEid() + " is pushing grades for course # " + canvasService.getCourseId() + " to Canvas");
+            ModelAndView page = new ModelAndView("forward:/courseConfiguration/" + sectionId);
+
+            Long courseId = Long.valueOf(canvasService.getCourseId());
+
+            boolean isSimpleAttendance = classSetupForm.getSimpleAttendance();
+            List<AttendanceSummaryModel> summaryForSections = isSimpleAttendance ?
+                    reportService.getSimpleAttendanceSummaryReport(sectionId) :
+                    reportService.getAviationAttendanceSummaryReport(sectionId);
+
+            AttendanceAssignment assignmentConfigurationFromSetup = generateAssignmentFromClassSetupForm(classSetupForm);
+            try {
+                assignmentSubmitter.submitCourseAttendances(isSimpleAttendance, summaryForSections, courseId, canvasService.getOauthToken(), assignmentConfigurationFromSetup);
+                page.addObject("pushingSuccessful", true);
+            }
+            catch (AttendanceAssignmentException e){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1166)